### PR TITLE
Split out compdata into its own package

### PR DESCRIPTION
**Pull request by pralitp**
Split out compdata into its own R package and load it in the old/new test as needed.

### CODE REVIEW

**Comment by pralitp** [comment]
Some clean up to remains.  I had a couple of questions though.  I made the new compdata off of the outputs of gcamdata v1.0.  Thus the flags for X years or no Long years are no longer needed?  Also do we still want to make the old/new tests optional (We would still get failures when people make changes that *should* change the outputs?

File: tests/testthat/test_oldnew.R [hunk 1]
```diff
@@ -50,6 +50,11 @@ test_that("matches old data system output", {
                        "GCAM_DATA_MAP precursors doesn't match. Rerun generate_package_data to update.")
   }
 
+  if(!requireNamespace("gcamdata.compdata", quietly = TRUE)) {
```
Comment (bpbond) [inline tests/testthat/test_oldnew.R]: Why not simply `require` here?

File: tests/testthat/test_oldnew.R [hunk 2]
```diff
@@ -50,6 +50,11 @@ test_that("matches old data system output", {
...
+    skip("gcamdata.compdata package not available unable to load COMPDATA")
```
Comment (bpbond) [inline tests/testthat/test_oldnew.R]: "gcamdata.compdata package not available" probably clearer and shorter

**Comment by bpbond** [comment]
>Thus the flags for X years or no Long years are no longer needed? 

Right!

>Also do we still want to make the old/new tests optional (We would still get failures when people make changes that should change the outputs?

Hmm. It seems like the old/new test should be required if you're doing development work?

**Comment by pralitp** [comment]
Ok, I've cleaned up the commented out code and made your suggested changes.

> Hmm. It seems like the old/new test should be required if you're doing development work?

Yea, maybe I'm jumping the gun on a problem that doesn't exist yet.  If we decide getting "failures" from expected changes is distracting and happens with enough frequency in the future we can put in some way of controlling its then.

**Pull request merged**